test/new-e2e/pkg/environments/aws/ecs/ecs.go

```diff
@@ -0,0 +1,290 @@
+func WithExtraConfigParams(configMap runner.ConfigMap) ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.extraConfigParams = configMap
+		return nil
+	}
+}
+
+// WithFakeIntakeOptions sets the options for the FakeIntake
+func WithFakeIntakeOptions(opts ...fakeintake.Option) ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.fakeintakeOptions = append(params.fakeintakeOptions, opts...)
+		return nil
+	}
+}
+
+// WithECSFargateCapacityProvider enable Fargate ECS
+func WithECSFargateCapacityProvider() ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.ecsFargate = true
+		return nil
+	}
+}
+
+// WithECSLinuxECSOptimizedNodeGroup enable aws/ecs/linuxECSOptimizedNodeGroup
+func WithECSLinuxECSOptimizedNodeGroup() ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.ecsLinuxECSOptimizedNodeGroup = true
+		return nil
+	}
+}
+
+// WithECSLinuxECSOptimizedARMNodeGroup enable aws/ecs/linuxECSOptimizedARMNodeGroup
+func WithECSLinuxECSOptimizedARMNodeGroup() ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.ecsLinuxECSOptimizedARMNodeGroup = true
+		return nil
+	}
+}
+
+// WithECSLinuxBottlerocketNodeGroup enable aws/ecs/linuxBottlerocketNodeGroup
+func WithECSLinuxBottlerocketNodeGroup() ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.ecsLinuxBottlerocketNodeGroup = true
+		return nil
+	}
+}
+
+// WithECSWindowsNodeGroup enable aws/ecs/windowsLTSCNodeGroup
+func WithECSWindowsNodeGroup() ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.ecsWindowsNodeGroup = true
+		return nil
+	}
+}
+
+// WithInfraShouldDeployFakeintakeWithLB enable load balancer on Fakeintake
+func WithInfraShouldDeployFakeintakeWithLB() ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.infraShouldDeployFakeintakeWithLB = true
+		return nil
+	}
+}
+
+// WithoutFakeIntake deactivates the creation of the FakeIntake
+func WithoutFakeIntake() ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.fakeintakeOptions = nil
+		return nil
+	}
+}
+
+// WithoutAgent deactivates the creation of the Docker Agent
+func WithoutAgent() ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.agentOptions = nil
+		return nil
+	}
+}
+
+// Run deploys a ECS environment given a pulumi.Context
+func Run(ctx *pulumi.Context, env *environments.ECS, params *ProvisionerParams) error {
+	var awsEnv aws.Environment
+	var err error
+	if env.AwsEnvironment != nil {
+		awsEnv = *env.AwsEnvironment
+	} else {
+		awsEnv, err = aws.NewEnvironment(ctx)
+		if err != nil {
+			return err
+		}
+	}
+	// Create cluster
+	ecsCluster, err := ecs.CreateEcsCluster(awsEnv, params.name)
+	if err != nil {
+		return err
+	}
+
+	// Export cluster’s properties
+	ctx.Export("ecs-cluster-name", ecsCluster.Name)
+	ctx.Export("ecs-cluster-arn", ecsCluster.Arn)
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
Exported properties could be part of the environment. Here these are exported as not typed, but the provisioner is typed which could be confusing. I can take care of adding the type.